server/src/main/java/org/opensearch/env/NodeEnvironment.java

```diff
@@ -342,6 +356,25 @@ public NodeEnvironment(Settings settings, Environment environment) throws IOExce
             }
             this.locks = nodeLock.locks;
             this.nodePaths = nodeLock.nodePaths;
+            this.cacheNodePath = nodePaths[0];
+
+            /*
+              The following block initializes the search cache based on user configuration.
+              If the user doesn't configure the cache size, we default to 100 MB since we still
```

Comment:
I think the ability to specify a percentage will be super useful here, similar to how it works with [the existing memory size setting](https://github.com/opensearch-project/OpenSearch/blob/ad7ce4cd446523d52286530ccfbb1dabc2dc7f86/server/src/main/java/org/opensearch/common/unit/MemorySizeValue.java). What do you think?

Reply:
From my observation, I noticed storage related settings to be defined as a fixed capacity instead of percentages. We could lead the path here, but there are a couple of things which will need clarification/user understanding. 
- Memory is a single path concept, which makes it easier to be defined in terms of percentages. 
- Currently, we support multiple paths for nodes, and pick the first one for search cache, and the setting in this case can mean different things -
- - % of total disk space
- - % of first, total data path space
- - % of available disk space
- - % of first, available data path space

I know that with #6274 , we will solve the multiple data path case but we may still need to clarify on available vs total, and the ever-changing size cache when the node restarts if go with a percentage of the available storage size.

Walking through all of the reasons, I resorted to a number instead of percentage. But I am open to thoughts 🙂

Reply:
I would think percentage of total would be the way to go. In any case we will want to support an absolute value, so I think it is fine to start here. We can always add the option of specifying a percentage later.

However, I do think it would be nice for a user not to have to think about this setting when deploying a dedicated search node. With no `data` role to contend with disk space, a default value of 100MB is obviously never going to be optimal. In that case defaulting to 90% of total disk space or something like that would be much better. What do you think?

Reply:
I'm thinking something like:

```
if setting has not been specified:
  if dedicated search node:
    use 90% of total space on the cache path disk
  else:
    fail node startup and make the user choose a value
```

Reply:
The only scenario where that would be a problem is when the user decides to add the `data` role post cache generation. In that case, we will need some cache shrinking logic when the node starts with added roles in. 
We will also need this logic in case of cache size change, where the user decides to reduce the cache size in some scenarios. 
I can think through this designing for the cache recovery mechanism as a part of #5980 .

Otherwise, the idea seems to have some additional logic on the lines of - 
- `if (node.isData && node.isSearch)` - reserve MBs as currently, but if data role was added now, shrink cache from % to MBs.
- `if (node.isSearch)` - reserve a higher % by default.

We will need to document this behavior to safeguard the user against unintended consequences.

Reply:
> I'm thinking something like:
> 
> ```
> if setting has not been specified:
>   if dedicated search node:
>     use 90% of total space on the cache path disk
>   else:
>     fail node startup and make the user choose a value
> ```

I was late thinking through and typing it in. That makes more sense - Simpler for the user to define and server to maintain.

Reply:
Updated the code and tests to reflect this behavior.